routing/endpointregistry_test.go

```diff
@@ -146,7 +226,7 @@ func benchmarkIncInflightRequests(b *testing.B, name string, goroutines int) {
 			go func() {
 				defer wg.Done()
 				for n := 0; n < b.N/goroutines; n++ {
-					r.IncInflightRequest(key)
+					r.GetMetrics(key).IncInflightRequest()
```

Comment:
Move r.GetMetrics(key) outside of the loop. 

We would put the pointer to the metrics into LBEndpoint so we need to understand performance and allocations of the IncInflightRequest() only.
Please also add benchmark for GetDetected and InflightRequests reading.

Reply:
Ah, I see, you would like to see benchmark for `Metrics.DetectedTime()` as well.

Reply:
Maybe it also makes sense to add a test that verifies that getters do not allocate (similar to TestOAuthTokeninfoAllocs in #2794)

Reply:
I see methods `IncInflightRequest`, `DecInflightRequest`, `InflightRequests`, `DetectedTime`, `LastSeen` do not allocate, created a test to check this fact.